File: matlab/add_control_noise.m
function [V,G]= add_control_noise(V,G,Q, addnoise)
% Add random noise to nominal control values

if addnoise == 1
%     V= V + randn(1)*sqrt(Q(1,1)); % if assume Q is diagonal
%     G= G + randn(1)*sqrt(Q(2,2));
    
     C= multivariate_gauss([V;G],Q, 1); % if Q might be correlated
     V= C(1); G= C(2);
end

File: cpp/add_control_noise.cpp
#include "add_control_noise.h"
#include <iostream>

using namespace std;

void add_control_noise(float &V, float &G, Matrix2f Q, int addnoise, float* VnGn) 
{
	if (addnoise ==1) {
		VectorXf A(2);
		A(0) = V;
		A(1) = G;
		VectorXf C(2);
		C = multivariate_gauss(A,Q,1);
		VnGn[0] = C(0);
		VnGn[1] = C(1);
	}
}
